tests/unit/Core/Checkout/Promotion/Cart/PromotionProcessorTest.php

```diff
@@ -126,4 +127,70 @@ public function testProcessWithCartZeroPriceAndPromotionIsNotGlobal(): void
         static::assertCount(1, $toCalculateCart->getErrors());
         static::assertInstanceOf(PromotionsOnCartPriceZeroError::class, $toCalculateCart->getErrors()->first());
     }
+
+    /**
+     * Test that promotions with considerAdvancedRules=true but uninitialized filter keys
+     * work correctly when adding items to cart without throwing initialization errors.
+     *
+     * This test simulates the real cart scenario where your targeted fix prevents crashes.
+     */
+    public function testProcessWithConsiderAdvancedRulesButUninitializedFilterKeys(): void
+    {
+        $promotionCalculatorMock = $this->createMock(PromotionCalculator::class);
+        $groupBuilderMock = $this->createMock(LineItemGroupBuilder::class);
+
+        $promotionProcessor = new PromotionProcessor($promotionCalculatorMock, $groupBuilderMock);
+
+        $originalCart = new Cart('test');
+        $toCalculateCart = new Cart('test');
+        $toCalculateCart->setPrice(new CartPrice(100, 100, 100, new CalculatedTaxCollection(), new TaxRuleCollection(), CartPrice::TAX_STATE_NET));
+
+        $context = $this->createMock(SalesChannelContext::class);
```

Comment:
Note that we try to avoid mocking the sales channel context, use this instead:
```suggestion
        $context = Generator::generateSalesChannelContext()
```